src/pages/cli/graphql/examples-and-solutions.mdx

```diff
@@ -0,0 +1,565 @@
+  orderTotal: Int
+  salesPeriod: String @index(name: "bySalesPeriodByOrderTotal", queryField: "repsByPeriodAndTotal", sortKeyFields: ["orderTotal"])
+}
+
+type Inventory @model {
+  productID: ID! @primaryKey(sortKeyFields: ["warehouseID"])
+  warehouseID: ID! @index(name: "byWarehouseID", queryField: "itemsByWarehouseID")
+  inventoryAmount: Int!
+}
+
+type Product @model {
+  id: ID!
+  name: String!
+  orders: [Order] @hasMany(indexName: "byProduct", fields: ["id"])
+  inventories: [Inventory] @hasMany(fields: ["id"])
+}
+```
+
+Now that we have the schema created, let's create the items in the database that we will be operating against:
+
+```graphql
+# first
+mutation createWarehouse {
+  createWarehouse(input: {id: "1"}) {
+    id
+  }
+}
+
+# second
+mutation createEmployee {
+  createEmployee(input: {
+    id: "amanda"
+    name: "Amanda",
+    startDate: "2018-05-22",
+    phoneNumber: "6015555555",
+    warehouseID: "1",
+    jobTitle: "Manager",
+    newHire: "true"}
+  ) {
+    id
+    jobTitle
+    name
+    newHire
+    phoneNumber
+    startDate
+    warehouseID
+  }
+}
+
+# third
+mutation createAccountRepresentative {
+  createAccountRepresentative(input: {
+    id: "dabit"
+    orderTotal: 400000
+    salesPeriod: "January 2019"
+  }) {
+    id
+    orderTotal
+    salesPeriod
+  }
+}
+
+# fourth
+mutation createCustomer {
+  createCustomer(input: {
+    id: "jennifer_thomas"
+    accountRepresentativeID: "dabit"
+    name: "Jennifer Thomas"
+    phoneNumber: "+16015555555"
+  }) {
+    id
+    name
+    accountRepresentativeID
+    phoneNumber
+  }
+}
+
+# fifth
+mutation createProduct {
+  createProduct(input: {
+    id: "yeezyboost"
+    name: "Yeezy Boost"
+  }) {
+    id
+    name
+  }
+}
+
+# sixth
+mutation createInventory {
+  createInventory(input: {
+    productID: "yeezyboost"
+    warehouseID: "1"
+    inventoryAmount: 300
+  }) {
+    productID
+    inventoryAmount
+    warehouseID
+  }
+}
+
+# seventh
+mutation createOrder {
+  createOrder(input: {
+    amount: 300
+    date: "2018-07-12"
+    status: "pending"
+    accountRepresentativeID: "dabit"
+    customerID: "jennifer_thomas"
+    productID: "yeezyboost"
+  }) {
+    id
+    customerID
+    accountRepresentativeID
+    amount
+    date
+    customerID
+    productID
+  }
+}
+```
+
+### 1. Look up employee details by employee ID
+
+This can simply be done by querying the employee model with an employee ID, no `@primaryKey` or `@index` need to be explicitly specified to make this work.
+
+```graphql
+query getEmployee($id: ID!) {
+  getEmployee(id: $id) {
+    id
+    name
+    phoneNumber
+    startDate
+    jobTitle
+  }
+}
+```
+
+### 2. Query employee details by employee name
+
+The `@index` `byName` on the `Employee` type makes this access-pattern feasible because under the covers an index is created and a query is used to match against the name field. We can use this query:
```

Comment:
Maybe worth noting that the query name is defined in the index, not auto generated. A query won't be created if the name isn't defined

Reply:
Discussed in data-modeling section already.